src/components/mainView/SingleRecord.tsx

```diff
@@ -1,31 +1,43 @@
-import React from 'react'
-import Row from 'react-bootstrap/Row'
-import Col from 'react-bootstrap/Col'
-import DropdownMenu from 'components/dropdownMenu/index'
-function SingleRecord() {
+import React from "react";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import DropdownMenu from "components/dropdownMenu/index";
+import { IStudentRaw } from "state/ducks/students/types";
+
+type SingleRecordProps = {
+  student: IStudentRaw;
+  handleEdit: Function;
+  handleDelete: Function;
+};
+
+const SingleRecord = ({
+  student,
+  handleEdit,
+  handleDelete,
+}: SingleRecordProps) => {
   return (
     <>
-      <Row style={{ marginTop: '20px' }}>
-        <Col> Ahmed </Col>
-        <Col> 96 </Col>
-        <Col> Math </Col>
+      <Row style={{ marginTop: "20px" }}>
+        <Col> {student?.name} </Col>
+        <Col> {student?.marks} </Col>
+        <Col> {student?.subject} </Col>
         <Col>
-          <div className="gradeBox" style={{ color: '#ffb59d' }}>
-            B+
+          <div className="gradeBox" style={{ color: "#ffb59d" }}>
+            {student?.grade}
           </div>
         </Col>
         <Col>
           <div> Jan 29, 2022 </div>
           <div className="timeStamp"> at 8:00 PM</div>
         </Col>
         <Col>
-          {' '}
-          <DropdownMenu />{' '}
+          {" "}
+          <DropdownMenu handleEdit={handleEdit} handleDelete={handleDelete} />
```

Comment:
Use proper container patterns. This is considered props drilling. in the container pattern, each component using either an action or redux state should have a container.